pocs/utils/data.py

```diff
@@ -1,46 +1,145 @@
+        self.wide_field = wide_field
+        self.narrow_field = narrow_field
+        self.keep_going = keep_going
+
+    def download_all_files(self):
+        """Downloads the files according to the attributes of this object."""
+        result = True
+        try:
+            download_IERS_A()
+        except Exception as e:
+            if not self.keep_going:
+                raise e
+            print('Failed to download IERS A bulletin: {}'.format(e))
+            result = False
+        if self.wide_field:
+            for i in range(4110, 4119):
+                if not self.download_one_file('4100/index-{}.fits'.format(i)):
+                    result = False
+        if self.narrow_field:
+            for i in range(4210, 4219):
+                if not self.download_one_file('4200/index-{}.fits'.format(i)):
+                    result = False
+        return result
+
+    def download_one_file(self, fn):
+        """Downloads one astrometry.net file into self.data_folder."""
+        dest = "{}/{}".format(self.data_folder, os.path.basename(fn))
+        if os.path.exists(dest):
+            return True
+        url = "http://data.astrometry.net/{}".format(fn)
+        try:
             df = data.download_file(url)
-            try:
-                shutil.move(df, dest)
-            except OSError as e:
-                print("Problem saving. (Maybe permissions?): {}".format(e))
+        except Exception as e:
+            if not self.keep_going:
+                raise e
+            print('Failed to download {}: {}'.format(url, e))
+            return False
+        try:
+            shutil.move(df, dest)
+            return True
+        except OSError as e:
+            if not self.keep_going:
+                raise e
+            print("Problem saving {}. Check permissions: {}".format(url, e))
+            return False
 
-    if wide_field:
-        for i in range(4110, 4119):
-            download_one_file('4100/index-{}.fits'.format(i))
 
-    if narrow_field:
-        for i in range(4210, 4219):
-            download_one_file('4200/index-{}.fits'.format(i))
+def download_all_files(data_folder=None, wide_field=True, narrow_field=False, keep_going=True):
+    dl = Downloader(
+        data_folder=data_folder,
+        keep_going=keep_going,
+        narrow_field=narrow_field,
+        wide_field=wide_field)
+    return dl.download_all_files()
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        '--folder',
+        help=('Destination folder for astrometry indices. Default: {}'.format(
+            _default_data_folder())),
+        default=_default_data_folder())
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--keep_going',
```

Comment:
We don't have a good convention on argument names but I think I prefer the dash, which gets converted to an underscore for the parameter. So `--keep-going` becomes `keep_going` as variable.

For some reason underscores in command line parameters seems off to me. Thoughts?